tests/test_reconciliation/test_base.py

```diff
@@ -0,0 +1,105 @@
+from unittest.mock import Mock
+
+import numpy as np
+import pandas as pd
+import pytest
+from scipy.sparse import csr_matrix
+
+from etna.datasets import TSDataset
+from etna.reconciliation.base import BaseReconciliator
+
+
+class DummyReconciliator(BaseReconciliator):
+    def fit(self, ts: TSDataset) -> "DummyReconciliator":
+        self.mapping_matrix = Mock()
+        return self
+
+
+@pytest.fixture
+def hierarchical_ts():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 4,
+            "market": ["X"] * 2 + ["X"] * 2 + ["Y"] * 2 + ["Y"] * 2,
+            "product": ["a"] * 2 + ["b"] * 2 + ["c"] * 2 + ["d"] * 2,
+            "target": [1.0, 2.0] + [3.0, 4.0] + [5.0, 10.0] + [15.0, 20.0],
+        }
+    )
+    df_exog = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03"] * 2,
+            "market": ["X"] * 3 + ["Y"] * 3,
+            "exog": [10.0, 12.0, 13.0] + [14.0, 15.0, 16.0],
+        }
+    )
+    df, hs = TSDataset.to_hierarchical_dataset(
+        df=df, level_columns=["market", "product"], keep_level_columns=False, return_hierarchy=True
+    )
+    df_exog, _ = TSDataset.to_hierarchical_dataset(
+        df=df_exog, level_columns=["market"], keep_level_columns=False, return_hierarchy=False
+    )
+    ts = TSDataset(df=df, freq="D", df_exog=df_exog, known_future=["exog"], hierarchical_structure=hs)
+    return ts
+
+
+@pytest.fixture
+def market_total_mapping_matrix():
+    mapping_matrix = np.array([[1, 1]])
+    mapping_matrix = csr_matrix(mapping_matrix)
+    return mapping_matrix
+
+
+@pytest.fixture
+def total_market_mapping_matrix():
+    mapping_matrix = np.array([[1 / 6], [5 / 6]])
+    mapping_matrix = csr_matrix(mapping_matrix)
+    return mapping_matrix
+
+
+@pytest.mark.parametrize("source_level", ("product", "market", "total"))
+def test_aggregate(hierarchical_ts, source_level):
+    reconciliator = DummyReconciliator(target_level="level", source_level=source_level)
+    ts_aggregated = reconciliator.aggregate(ts=hierarchical_ts)
+    assert ts_aggregated.current_df_level == source_level
+
+
+def test_aggregate_fails_low_source_level(hierarchical_ts):
+    ts_market_level = hierarchical_ts.get_level_dataset(target_level="market")
+    reconciliator = DummyReconciliator(target_level="level", source_level="product")
+    with pytest.raises(ValueError, match=""):
```

Comment:
Can't we check error message more specifically?

Reply:
My fall)